src/MainMenu.cpp

```diff
@@ -1769,4 +1769,256 @@ ZunResult MainMenu::ChoosePracticeLevel()
 }
 #pragma optimize("", on)
 
+#pragma optimize("s", on)
+#pragma var_order(i, optionsVm)
+u32 MainMenu::DrawOptionsMenu()
+{
+
+    AnmVm *optionsVm;
+    i32 i;
+
+    MoveCursor(this, 9);
+    optionsVm = &this->vm[8];
+    for (i = 0; i < 9; i++)
+    {
+        if ((i >= 5) && (i <= 7))
+        {
+            this->setSavedCursorPosition(&this->vm[i + 0x43], i, i, this->cursor);
+        }
+        else
+        {
+            this->setSavedCursorPosition(optionsVm, i, i, this->cursor);
+            optionsVm++;
+        }
+    }
+
+    for (i = 0; i < 5; i++, optionsVm++)
+    {
+        this->setSavedCursorPosition(optionsVm, CURSOR_OPTIONS_POS_LIFECOUNT, i, g_Supervisor.cfg.lifeCount);
+    }
+
+    for (i = 0; i < 4; i++, optionsVm++)
+    {
+        this->setSavedCursorPosition(optionsVm, CURSOR_OPTIONS_POS_BOMBCOUNT, i, g_Supervisor.cfg.bombCount);
+    }
+    for (i = 0; i < 2; i++, optionsVm++)
+    {
+        this->setSavedCursorPosition(optionsVm, CURSOR_OPTIONS_POS_COLORMODE, i, g_Supervisor.cfg.colorMode16bit);
+    }
+    for (i = 0; i < 2; i++, optionsVm++)
+    {
+        this->setSavedCursorPosition(optionsVm, CURSOR_OPTIONS_POS_PLAYSOUNDS, i, g_Supervisor.cfg.playSounds);
+    }
+    optionsVm = &this->vm[0x4d];
+
+    for (i = 0; i < 3; i++, optionsVm++)
+    {
+        this->setSavedCursorPosition(optionsVm, CURSOR_OPTIONS_POS_MUSICMODE, i, g_Supervisor.cfg.musicMode);
+    }
+    optionsVm = &this->vm[0x4b];
```

Comment:
```suggestion
    optionsVm = &this->vm[75];
```

Reply:
Should i use hex on the next if or not?
```diff 
- if (this->stateTimer >= 0x20)
+ if (this->stateTimer >= 32)
```

Reply:
Yeah, I think that makes more sense. stateTimer is a frame counter, I don't think using hex for those make much sense.